README.md

```diff
@@ -2,4 +2,114 @@
 
 ## Marine Rust SDK
 
-This SDK is intended to run backend application on the Fluence network with [Marine](https://github.com/fluencelabs/marine) runtime. More information about usage and some internals could found in [docs](https://fluence.dev/docs/marine-book/marine-rust-sdk/).
+This SDK empowers developers to create suitable general-purpose Wasm services executable by the [Marine](https://github.com/fluencelabs/marine) runtime. Such services are constructed from one or more Wasm modules with the help of [interface-types](https://github.com/WebAssembly/interface-types) and a [shared-nothing linking](https://training.linuxfoundation.org/blog/how-and-why-to-link-webassembly-modules/) scheme. Each module results from Rust code compiled to `wasm32-wasi` compile target with some glue code auto-generated by this SDK. The primary purpose of the SDK influenced its design is to provide developers a way to write Wasm modules on "vanilla" Rust, wrapping necessary blocks with the supplied macros.
+
+
+## Usage
+
+The core component of the SDK is the #[marine] macro that should be used with export functions, extern blocks, and structures. Let's consider a simplest scenario with a module with one export function:
+```rust
+#[marine]
+use marine_rs_sdk::marine;
+use marine_rs_sdk::module_manifest;
+
+#[marine]
+pub fn greeting(name: String) -> String {
+    format!("Hi, {}", name)
+}
+```
+This code imports the Marine SDK and wraps the greeting function with the #[marine] macro. Every function wrapped in such way will be exported from a compiled Wasm module.
+
+To build this module into Wasm, you'll be needed the marine CLI tool (`cargo install marine` to install):
+
+```bash
+marine build --release
+```
+
+Finally, you could obtain some build info from the resulted binary and play with it with the help of mrepl (`cargo install mrepl` to install):
```

Comment:
how about:
Finally, you received some build info and are now in a position to play with it your Wasm modules with the help of Marine RELP (mrepl). if you don't have it, install with `cargo install mrepl`:

Reply:
I make it unclear; by `build info` I meant info embedded into a Wasm binary, and the following `marine info` command is intended to show this build info. How about putting it in the following way:

```
Finally, you could obtain a build manifest embedded into a Wasm binary and play with a module with the help of Marine REPL (mrepl). If you don't have it, install it with `cargo install mrepl`:
```

Reply:
close :) 
Finally, you obtain a build manifest embedded into a Wasm binary and can interact with a module in the Marine REPL (mrepl). If you don't have it, install it with `cargo install mrepl`:

should we add a link-out to mrepl?